fhd_core/source_modeling/source_array_import.pro

```diff
@@ -0,0 +1,87 @@
+FUNCTION source_array_import,file_path=file_path,no_extend=no_extend,$
+    id_column=id_column,x_column=x_column,y_column=y_column,ra_column=ra_column,dec_column=dec_column,$
+    ston_column=ston_column,frequency_column=frequency_column,flag_colum=flag_column,parent_column=parent_column,$
+    flux_xx_column=flux_xx_column,flux_yy_column=flux_yy_column,$
+    flux_xy_column=flux_xy_column,flux_yx_column=flux_yx_column,$
+    flux_I_column=flux_I_column,flux_Q_column=flux_Q_column,$
+    flux_U_column=flux_U_column,flux_V_column=flux_V_column,_Extra=extra
+
+textfast,source_array,header,first_line=1,file_path=file_path,/read
+IF N_Elements(id_column) EQ 0 THEN id_column=(where(header EQ '#id'))[0]
+IF N_Elements(x_column) EQ 0 THEN x_column=(where(header EQ 'x_loc'))[0]
+IF N_Elements(y_column) EQ 0 THEN y_column=(where(header EQ 'y_loc'))[0]
+IF N_Elements(ra_column) EQ 0 THEN ra_column=(where(header EQ 'RA'))[0]
+IF N_Elements(dec_column) EQ 0 THEN dec_column=(where(header EQ 'Dec'))[0]
+IF N_Elements(ston_column) EQ 0 THEN ston_column=(where(header EQ 'S/N'))[0]
+IF N_Elements(frequency_column) EQ 0 THEN frequency_column=(where(header EQ 'frequency'))[0]
+IF N_Elements(flag_column) EQ 0 THEN flag_column = (where(header EQ 'Flag'))[0]
+IF N_Elements(parent_column) EQ 0 THEN parent_column = (where(header EQ 'Parent'))[0]
+
+IF N_Elements(flux_xx_column) EQ 0 THEN flux_xx_column=(where(header EQ 'XX_apparent'))[0]
+IF N_Elements(flux_yy_column) EQ 0 THEN flux_yy_column=(where(header EQ 'YY_apparent'))[0]
+IF N_Elements(flux_xy_column) EQ 0 THEN flux_xy_column=(where(header EQ 'XY_apparent'))[0]
+IF N_Elements(flux_yx_column) EQ 0 THEN flux_yx_column=(where(header EQ 'YX_apparent'))[0]
+IF N_Elements(flux_I_column) EQ 0 THEN flux_I_column=(where(header EQ 'Stokes_I'))[0]
+IF N_Elements(flux_Q_column) EQ 0 THEN flux_Q_column=(where(header EQ 'Stokes_Q'))[0]
+IF N_Elements(flux_U_column) EQ 0 THEN flux_U_column=(where(header EQ 'Stokes_U'))[0]
+IF N_Elements(flux_V_column) EQ 0 THEN flux_V_column=(where(header EQ 'Stokes_V'))[0]
+
```

Comment:
In IDL 8.x, a -1 returned from a where statement can give interesting results with array wrapping. I did just notice the other day that there's now a NULL keyword to where (in IDL 8.x) that tells where to return a NULL if no condition matches, rather than a -1.